kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/WifiAccessPointImpl.java

```diff
@@ -141,4 +142,82 @@ public String toString() {
         }
         return sb.toString();
     }
+    
+    @Override
+	public int hashCode() {
```

Comment:
Yeah, you are right. Equals must be consistent with hashcode to have it work with hash-based data structures. The IDE generated implementations of these `equals` and `hashcode` methods are verbose and expose low-level complexities. That's why in Java 7, few useful static factory methods are provided to write your own `equals` and `hashcode` implementation in a cleaner way. Have a look at the `hash(Object...)` and `equals(Object, Object)` methods in `Objects`.